### PR TITLE
Fixes ordering key and improves rejection

### PR DESCRIPTION
**Pull request by mickprev**
Q | A
-- | --
Bug fix? | yes
New feature? | yes
BC breaks? | yes
Deprecations? | no
Tickets | ~
License | MIT

Currently, even if the code takes into account the "Ordering Key", it's not really sent to Google PUB / SUB because the setOrderingKey method in MessageBuilder class returns a new instance of Message and does not modify the current message.

Next, In case of rejection of a message, we want to notify Google PUB / SUB right away and not 10 seconds later, so I have filled the reject method.

So this PR: 

- Fixes the problems on "Ordering Key".
- Adds the reject feature. There is therefore a BC break, we now wait less than 1 second (versus minimum 10 secodes before) for informing PUB / SUB that the message has failed. I was inspired by this documentation: https://googleapis.dev/python/pubsub/latest/subscriber/index.html
- Adds the ".editorconfig" file for consistent coding styles between different editors
- Adds the ".gitattributes" file to exclude some files and folders when compressing
- Updates the ".gitignore" file. The ".idea" file is specific to PHPStorm, the exclusion must be in a gitignore specific to the environment and not to the project.
- Adds PHP Coding Standards Fixer tool to follow standards. Command to execute: vendor/bin/php-cs-fixer fix
- Adds PHPStan for focuses on finding errors in the code without actually running it. Command to execute: vendor/bin/phpstan analyse Tests Transport --level=5
- Adds PHPUnit for unit testing. I respected the "boy scout rule" by adding some tests. Command to execute: vendor/bin/phpunit

Later:

- It would be necessary to execute in a CI the 3 commands described above.
- Increase the level of PHPStan.

### CODE REVIEW

**Comment by mickprev** [comment]
Ping @marforon

File: Transport/GpsReceiver.php [hunk 1]
```diff
@@ -1,118 +1,127 @@
...
+            ;
+        } catch (Throwable $exception) {
+            throw new TransportException($exception->getMessage(), 0, $exception);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * It does nothing. We can't reject message in Google Pub/Sub.
```
Comment (marforon) [inline Transport/GpsReceiver.php:77]: Could you please alter this comment? I would keep the reference to the retry policy, but this part about doing nothing is not relevant anymore :).

**Review by marforon — CHANGES_REQUESTED** [review]
Thanks for your contribution @mickprev! Great work!
I think you commited composer.lock by mistake. I will have to setup CI for this repo, but it's perfect you have time to do some tests.

File: Transport/GpsReceiver.php (hunk 1 above)
Comment (mickprev) [reply Transport/GpsReceiver.php:77]: I deleted the comment after my review and forgot to commit ... Sorry, I'm doing this right now :)

Comment (mickprev) [reply Transport/GpsReceiver.php:77]: I will also delete the composer.lock

**Review by marforon — APPROVED** [review]
Thank you @mickprev!

**Pull request merged**